Update LSP16 Specification (#211)
lukso-network/LIPs: LSPs/LSP-16-UniversalFactory.md

> `keccak256(abi.encodePacked(true, initializeCalldata, providedSalt))`

MUST emit the [ContractCreated] event after deploying the contract.

**Parameters:**

- `byteCode`: The bytecode of the contract to deploy                                                                                                  
- `providedSalt`: The salt provided by the deployer, which will be used to generate the final salt that will be used by the `CREATE2` opcode for contract deployment                                     
- `initializeCalldata`: The calldata to be executed on the created contract
- `constructorMsgValue`: The value sent to the contract during deploymentcontract                                                                                           
- `initializeCalldataMsgValue`: The value sent to the contract during initialization                                                                                               

**Return:**

- `contractCreated`: The address of the contract created.

**Requirements:**

> - If value is associated with the contract creation, the constructor of the contract to deploy MUST be payable, otherwise the call will revert.

Reviewer: suggested changes:

```suggestion
- If some value is transferred during the contract creation, the constructor of the contract to deploy MUST be payable, otherwise the call will revert.
```

> - If value is associated with the initialization call, the initialize function called on the contract to deploy MUST be payable, otherwise the call will revert.

Reviewer: suggested changes:

```suggestion
- If some value is transferred during the initialization call, the initialize function called on the contract to deploy MUST be payable, otherwise the call will revert.
```

- The sum of `constructorMsgValue` and `initializeCalldataMsgValue` MUST be equal to the value associated with the function call.

- MUST NOT use the same `bytecode`, `providedSalt` and `initializeCalldata` twice, otherwise the call will revert.

#### deployERC1167Proxy

```solidity
function deployERC1167Proxy(address implementationContract, bytes32 providedSalt) public returns (address);
```

Deploys an ERC1167 minimal proxy contract using the CREATE2 opcode.

> The address where the contract will be deployed can be known in advance via the {computeERC1167Address} function.

Reviewer: I suggest for these functions to use anchor links via `[computeERC1167Address](#computeERC1167Address)` instead of `{computeERC1167Address}`

> The `providedSalt` parameter is not used directly as the salt by the CREATE2 opcode. Instead, it is hashed via keccak256 with prepending a false boolean. See {generateSalt} function for more details.

Reviewer: same here
